Resources/Prototypes/Entities/Structures/Specific/anomalies.yml

```diff
@@ -231,3 +231,157 @@
     releasedGas: 8 # Frezon. Please replace if there is a better way to specify this
     releaseOnMaxSeverity: true
     spawnRadius: 0
+
+- type: entity
+  id: AnomalyLiquid
+  parent: BaseAnomaly
+  suffix: Liquid
+  components:
+  - type: Sprite
+    sprite: Structures/Specific/Anomalies/liquid_anom.rsi
+    layers:
+    - state: anom
+      map: ["enum.AnomalyVisualLayers.Base"]
+    - state: pulse
+      map: ["enum.AnomalyVisualLayers.Animated"]
+      visible: false
+  - type: Anomaly
+    anomalyContactDamage:
+      types:
+        Slash: 10
+  - type: PointLight
+    radius: 4.0
+    energy: 8.5
+    color: "#bbbbbb"
+  - type: Fixtures
+    fixtures:
+      fix1:
+        density: 20
+        mask:
+          - None
+        layer:
+          - Opaque
+  - type: ExplosionAnomaly
+    supercriticalExplosion: Default
+    explosionTotalIntensity: 50
+    explosionDropoff: 2
+    explosionMaxTileIntensity: 10
+  - type: ChasingAnomaly
+    chasingComponent: InjectableSolution
+  - type: LiquidAnomaly
+    possibleChemicals:
+    - Mold
```

Comment:
this will be impossible to maintain either have it be a blacklist or do it procedurally by summing possible damage of X units from the reagent metaboliser effects

Reply:
Do you suggest that when adding new reagents to the game, they also appear in the list of anomaly reagents? With blacklist support to manually remove some reagents

Did I understand correctly?

Reply:
just make it able to roll *any* chemical that is not in the blacklist

Reply:
can you tell me how to get all the chemicals of a certain type of prototype? is there such a ready-made function?

Reply:
theres a thing on IPrototypeManager to enumerate all prototypes

Reply:
A small game designer question. I've been thinking for a long time about the concept of an anomaly that can produce any substance in the game. The problem with this concept is that 80% of the substances that an anomaly can take in will not bring a significant gaming effect. The anomaly will be completely safe in most cases.

It is either necessary to make sure that within any available substance, the anomaly is still dangerous, somehow modifying the system and the lists of reagents. Or go back to the option with a white list, and manually configure a narrow list of interesting reagents that the anomaly will produce.